### PR TITLE
feat: nomic embed multimodal rag demo 

### PR DESCRIPTION
**Pull request by zanussbaum**
# Description

Adds Nomic Embed Multimodal demo
Also fixes bugs in NomicEmbedding class

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
@zanussbaum looks like the newer version of nomic isn't published yet?

**Comment by ravi03071991** [comment]
@zanussbaum small error

` def _get_imag_embeddings(self, images: List[ImageType]) -> List[List[float]]:
        return self._embed_image(images)`

it should be `_get_image_embeddings`.

File: llama-index-integrations/embeddings/llama-index-embeddings-nomic/llama_index/embeddings/nomic/base.py [hunk 1]
```diff
@@ -129,7 +134,7 @@ async def _aget_image_embedding(self, image: ImageType) -> List[float]:
     def _get_imag_embeddings(self, images: List[ImageType]) -> List[List[float]]:
```
Comment (ravi03071991) [inline llama-index-integrations/embeddings/llama-index-embeddings-nomic/llama_index/embeddings/nomic/base.py:134]: should be `_get_image_embeddings`

Comment (zanussbaum) [reply llama-index-integrations/embeddings/llama-index-embeddings-nomic/llama_index/embeddings/nomic/base.py]: oops thank you

**Review by ravi03071991 — APPROVED** [review]
LGTM.

**Review by Allzz77123 — APPROVED** [review]

**Pull request merged**